Crash with distinct instance mode

Given the config:

```yaml
server:
  http_listen_port: 9090
  log_level: debug
prometheus:
  wal_directory: /tmp/wal
  global:
    scrape_interval: 5s
  configs:
  - name: app
    scrape_configs:
      - job_name: app
        scrape_interval: 5s
        static_configs:
          - targets: ['app:3030']
    remote_write:
      - url: http://cortex:9009/api/prom/push
  instance_mode: distinct
integrations:
  agent:
    enabled: true
  node_exporter:
    enabled: true
    procfs_path: /host/proc
    sysfs_path: /host/sys
    rootfs_path: /rootfs
  prometheus_remote_write:
      - url: http://cortex:9009/api/prom/push
```

Running the agent results in a crash with this crash stack:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x793637]

goroutine 515 [running]:
regexp.(*Regexp).FindStringSubmatchIndex(0x0, 0xc000a7c2e0, 0xe, 0x8, 0x0, 0xc000a7c2e0)
/usr/local/go/src/regexp/regexp.go:1058 +0x37
github.com/prometheus/prometheus/pkg/relabel.relabel(0xc0015eb800, 0x8, 0x8, 0xc00028c880, 0xc000d46930, 0x1f9197f, 0x317ca00)
/src/agent/vendor/github.com/prometheus/prometheus/pkg/relabel/relabel.go:212 +0x8e5
github.com/prometheus/prometheus/pkg/relabel.Process(0xc0015eb800, 0x8, 0x8, 0xc000134d08, 0x1, 0x1, 0xc0015eb800, 0x8, 0x8)
/src/agent/vendor/github.com/prometheus/prometheus/pkg/relabel/relabel.go:185 +0x72
github.com/prometheus/prometheus/storage/remote.(*QueueManager).StoreSeries(0xc001610000, 0xc0015e0000, 0x2db, 0x400, 0x0)
/src/agent/vendor/github.com/prometheus/prometheus/storage/remote/queue_manager.go:531 +0x156
github.com/prometheus/prometheus/tsdb/wal.(*Watcher).readSegment(0xc000a4e0e0, 0xc001710000, 0x0, 0x1, 0x3126901, 0xc00011a050)
/src/agent/vendor/github.com/prometheus/prometheus/tsdb/wal/watcher.go:480 +0x78b
github.com/prometheus/prometheus/tsdb/wal.(*Watcher).watch(0xc000a4e0e0, 0x0, 0x1, 0x0, 0x0)
/src/agent/vendor/github.com/prometheus/prometheus/tsdb/wal/watcher.go:410 +0x415
github.com/prometheus/prometheus/tsdb/wal.(*Watcher).Run(0xc000a4e0e0, 0xb5946cf, 0x43a88c0)
/src/agent/vendor/github.com/prometheus/prometheus/tsdb/wal/watcher.go:250 +0x707
github.com/prometheus/prometheus/tsdb/wal.(*Watcher).loop(0xc000a4e0e0)
/src/agent/vendor/github.com/prometheus/prometheus/tsdb/wal/watcher.go:199 +0x138
created by github.com/prometheus/prometheus/tsdb/wal.(*Watcher).Start
/src/agent/vendor/github.com/prometheus/prometheus/tsdb/wal/watcher.go:173 +0x199
```

Removing `instance_mode: distinct` fixes the issue.

Credit to @hoenn for finding this.

--- a/pkg/integrations/manager.go
+++ b/pkg/integrations/manager.go
@@ -97,6 +97,8 @@ func (c *Config) ApplyDefaults() error {
 				{
 					SourceLabels: model.LabelNames{model.InstanceLabel},
 					Action:       relabel.Replace,
+					Separator:    ";",
+					Regex:        relabel.MustNewRegexp("(.*)"),
 					Replacement:  replacement,
 					TargetLabel:  model.InstanceLabel,
 				},

--- a/pkg/integrations/manager_test.go
+++ b/pkg/integrations/manager_test.go
@@ -13,9 +13,42 @@ import (
 	"github.com/grafana/agent/pkg/prom/instance"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	prom_config "github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/relabel"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/atomic"
+	"gopkg.in/yaml.v3"
 )
+
+func TestConfig_InstanceRelabels(t *testing.T) {
+	cfgText := `
+agent: 
+  enabled: true
+prometheus_remote_write:
+- url: http://localhost:9090/api/prom/push`
+
+	var (
+		cfg        Config
+		listenPort int = 12345
+	)
+	require.NoError(t, yaml.Unmarshal([]byte(cfgText), &cfg))
+
+	// Listen port must be set before applying defaults. Normally applied by the
+	// config package.
+	cfg.ListenPort = &listenPort
+	require.NoError(t, cfg.ApplyDefaults())
+
+	// Ensure that the relabel configs are functional
+	require.Len(t, cfg.PrometheusRemoteWrite, 1)
+	require.Len(t, cfg.PrometheusRemoteWrite[0].WriteRelabelConfigs, 1)
+
+	rules := cfg.PrometheusRemoteWrite[0].WriteRelabelConfigs[0]
+
+	result := relabel.Process(labels.FromStrings("instance", "127.0.0.1"), rules)
+
+	expectHostname, _ := instance.Hostname()
+	require.Equal(t, result.Get("instance"), expectHostname+":12345")
+}
 
 // TestManager_ValidInstanceConfigs ensures that the instance configs
 // applied to the instance manager are valid.

